### PR TITLE
[FIX] Kafka signal connector emits watermarks

### PR DESCRIPTION
**Pull request by mproch**

### CODE REVIEW

File: engine/flink/kafka-util/src/main/scala/pl/touk/nussknacker/engine/flink/util/signal/KafkaSignalStreamConnector.scala [hunk 1]
```diff
@@ -13,7 +14,11 @@ trait KafkaSignalStreamConnector {
   def connectWithSignals[A, B: TypeInformation](start: DataStream[A], processId: String, nodeId: String, schema: DeserializationSchema[B]): ConnectedStreams[A, B] = {
     val signalsSource = new FlinkKafkaConsumer[B](signalsTopic, schema,
       KafkaEspUtils.toProperties(kafkaConfig, Some(s"$processId-$nodeId-signal")))
-    val signals = start.executionEnvironment.addSource(signalsSource).name(s"signals-$processId-$nodeId")
+    val signals = start.executionEnvironment
+      .addSource(signalsSource).name(s"signals-$processId-$nodeId")
+      //We use ingestion time here, to advance watermarks in connected streams
+      //TODO: this is not always optimal solution, as e.g. in tests periodic watermarks are not the best option
+      .assignTimestampsAndWatermarks(new IngestionTimeExtractor[B])
```
Comment (arkadius) [inline engine/flink/kafka-util/src/main/scala/pl/touk/nussknacker/engine/flink/util/signal/KafkaSignalStreamConnector.scala]: Can you extract this assigner creation to protected method? Would be nice to have ability to override it. Some examples of assigner that i would see here:
- some punctuated assigner for tests
- assigner taking timestamp from signal event or at least kafka insertion timestamp (`previousElementTimestamp`)
The default one - ingenstion time is fine for me :+1:

File: engine/flink/process/src/test/scala/pl/touk/nussknacker/engine/process/helpers/ProcessTestHelpers.scala [hunk 2]
```diff
@@ -90,18 +111,19 @@ object ProcessTestHelpers {
...
+        WithCategories(new TestProcessSignalFactory(kafkaConfig, signalTopic)))
 
       override def buildInfo(): Map[String, String] = Map.empty
     }
 
     def invokeWithKafka(process: EspProcess, kafkaConfig: KafkaConfig,
                         config: Configuration = new Configuration(),
                         processVersion: ProcessVersion = ProcessVersion.empty,
-                        parallelism: Int = 1): JobExecutionResult = {
+                        parallelism: Int = 1)(action: => Unit): Unit = {
```
Comment (arkadius) [inline engine/flink/process/src/test/scala/pl/touk/nussknacker/engine/process/helpers/ProcessTestHelpers.scala]: Can you name this `action` more explicitly? It is kind of `postRunAction`?

**Comment by arkadius** [comment]
Some reported, little improvements would be nice :-)

File: engine/flink/process/src/test/scala/pl/touk/nussknacker/engine/process/helpers/ProcessTestHelpers.scala (hunk 2 above)
Comment (mproch) [reply engine/flink/process/src/test/scala/pl/touk/nussknacker/engine/process/helpers/ProcessTestHelpers.scala]: I change names to more informative

File: engine/flink/kafka-util/src/main/scala/pl/touk/nussknacker/engine/flink/util/signal/KafkaSignalStreamConnector.scala (hunk 1 above)
Comment (mproch) [reply engine/flink/kafka-util/src/main/scala/pl/touk/nussknacker/engine/flink/util/signal/KafkaSignalStreamConnector.scala]: Good idea, I extracted to method and added more docs

**Pull request merged**